### PR TITLE
EVG-5298 Implement CreateUserToken

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: ldap/ldap.go [hunk 1]
```diff
@@ -168,22 +186,75 @@ func (u *userService) isMemberOf(user, group string) error {
...
+			user.Name = entry.Values[0]
+		}
+		if entry.Name == "mail" {
+			user.EmailAddress = entry.Values[0]
+		}
+	}
+	return user
+}
+
+type User struct {
```
Comment (tychoish) [inline ldap/ldap.go]: There's a basic user in the gimlet package that you should be able to use without needing to do thise.

File: ldap/ldap_test.go [hunk 2]
```diff
@@ -212,3 +233,18 @@ func (s *LDAPSuite) TestGetUserByToken() {
...
+
+	token, err = s.um.CreateUserToken("nosuchuser", "")
+	s.Error(err)
+
+	token, err = s.um.CreateUserToken("erh", "hunter2")
+	s.NoError(err)
+	s.Equal("123456", token)
+	s.Equal("erh", mockPutUser.Username())
+	s.Equal("Eliot Horowitz", mockPutUser.DisplayName())
+	s.Equal("erh@mongodb.com", mockPutUser.Email())
```
Comment (tychoish) [inline ldap/ldap_test.go]: really?

File: ldap/ldap.go (hunk 1 above)
Comment (bsamek) [reply ldap/ldap.go]: Done.

File: ldap/ldap_test.go (hunk 2 above)
Comment (bsamek) [reply ldap/ldap_test.go]: YES

Comment (tychoish) [reply ldap/ldap_test.go]: no.....

**Pull request merged**